Fix project update

t0mk opened the pull request:
This PR fixes update logic for project resource, and adds payment_method_id attribute (and proper handling for it) to the project resource.

fixes #44

paultyng commented on packet/resource_packet_project.go:
```diff
@@ -1,10 +1,35 @@
 	"github.com/packethost/packngo"
 )
 
+const uuidRE = "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$"
```
I would personally do the `MustCompile` here and just assign it to a var:

```golang
var uuidRE = regexp.MustCompile(...)
```

As it will panic earlier just in case there is a problem.

t0mk replied:
thanks. I fixed it.

paultyng commented on packet/resource_packet_project.go:
```diff
@@ -27,6 +52,12 @@ func resourcePacketProject() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"payment_method_id": &schema.Schema{
```
You'll probably want to make this case insensitive for diffing?

paultyng replied:
Here is an example using `DiffSuppressFunc`: https://github.com/terraform-providers/terraform-provider-google/pull/1141/files

t0mk replied:
good idea, thanks, I did that. However, I can't add test for the billing methods, we don't have mechanism in place to use dummy ones.

t0mk commented:
@paultyng Thanks for your review. I bumped TF, so that I can use the RE StringMatch from Terraform.helpers.validation; and then I rebased this PR. I also fixed your comments from the review.
